rfc/wallpapers_v2_network_improvements.md

```diff
@@ -0,0 +1,69 @@
+* Start date: 2022-06-21
+* RFC PR: #TODO
+
+---
+
+## Summary
+
+The wallpapers feature was introduced on Android and iOS in v98 during Q1 2022. 
+Due to late scope changes, a networking solution was quickly put together
+in allow for delivering wallpapers from a remote source. As part of a
+larger set of improvements planned for Q3 2022, it is intended to revisit
+this implementation to create a more scalable version.
+
+---
+
+## Motivation
+
+The original version of the networking solution consisted of
+adding wallpaper image files to deeply nested directories which defined the properties
+of the wallpaper. For example, an Android wallpaper might be found at:
+```
+<root>/mobile-wallpapers/android/<resolution>/<orientation>/<light/dark>/<wallpaper collection>/<wallpaper name>
+```
+
+This directory structure was confusing and difficult to manage. Copying several versions of a file to different locations
+was painful, even when automated. Automation would require consistent file naming for each new set of wallpapers, which can
+be hard to guarantee. 
+
+Downloads were started automatically during app startup if a wallpaper was missing.
+
+The files were also hosted by another Mozillian team, which may introduce complexity in terms of ownership or cost.
+
+In general, we would like to achieve the following goals:
+
+- it is easier for product, UX, and ENG to add new wallpapers
+- it is possible to determine important metadata about a wallpaper, like its size or average color
+- download is transparent to the user
+- download is activated by the user
+- require less bandwidth, if possible
+- hosted in a more permanent, team-controlled location
+
+---
+
+## Guide-level explanation
+
+Instead of relying on an (undocumented) convention for file paths, wallpaper metadata will be contained in a top-level JSON schema.
+Using a well-known format will help discoverability, and JSON's extensiblility should allow for simpler long-term maintenance. 
```

Comment:
I would not use the word "extensiblility" but rather "flexible" or "unstructured". This is important to note - see comment below at drawbacks.